### PR TITLE
Fix displaying of global checkboxes in user_saml settings

### PR DESCRIPTION
**Pull request by KoffeinKaio**
Partly fix problems lined out in #573 
Templated wanted the values in config array, config array is only idp settings from user_saml, not the global ones.
Hope this approach is ok.

### CODE REVIEW

File: lib/Settings/Admin.php [hunk 1]
```diff
@@ -186,10 +186,17 @@ public function getForm() {
 
 		$type = $this->config->getAppValue('user_saml', 'type');
 		if ($type === 'saml') {
+			$generalSettings['require_provisioned_account'] = [
+                                'text' => $this->l10n->t('Only allow authentication if an account exists on some other backend (e.g. LDAP).', [$this->defaults->getName()]),
+                                'type' => 'checkbox',
+                                'global' => true,
+                                'value' => $this->config->getAppValue('user_saml', 'general-require_provisioned_account', 0)
+                        ];
```
Comment (CarlSchwan) [inline lib/Settings/Admin.php:194]:
```suggestion
				'text' => $this->l10n->t('Only allow authentication if an account exists on some other backend (e.g. LDAP).', [$this->defaults->getName()]),
				type' => 'checkbox',
				'global' => true,
				'value' => $this->config->getAppValue('user_saml', 'general-require_provisioned_account', 0)
			];
```

tab vs space ;)

**Review by blizzz — CHANGES_REQUESTED** [review]
there is something wrong with this PR.  Certainly the change in l10n should not be here.

**Comment by KoffeinKaio** [comment]
ah heck, I made a mistake by rebasing. I instead merged all new commits into my repo. lemme think about how to fix this

File: lib/Settings/Admin.php [hunk 2]
```diff
@@ -186,10 +186,17 @@ public function getForm() {
 
 		$type = $this->config->getAppValue('user_saml', 'type');
 		if ($type === 'saml') {
+			$generalSettings['require_provisioned_account'] = [
+				'text' => $this->l10n->t('Only allow authentication if an account exists on some other backend (e.g. LDAP).', [$this->defaults->getName()]),
+				type' => 'checkbox',
```
Comment (blizzz) [inline lib/Settings/Admin.php:191]:
```suggestion
				'type' => 'checkbox',
```

**Comment by KoffeinKaio** [comment]
aw to hell with that. lemme fix this again

**Comment by blizzz** [comment]
@KoffeinKaio i took the freedom to rebase, drop the unneeded commits and amend the missing single quote fix to your main commit

**Comment by KoffeinKaio** [comment]
> @KoffeinKaio i took the freedom to rebase, drop the unneeded commits and amend the missing single quote fix to your main commit

thanks, dont PR and work on other things my kids

**Comment by KoffeinKaio** [comment]
nice one, just wanted to suggest adding another func to SAMLSettings as I'm not THAT familiar with PHPUnit. Will add that to my todo list.

**Comment by blizzz** [comment]
> 



> nice one, just wanted to suggest adding another func to SAMLSettings as I'm not THAT familiar with PHPUnit. Will add that to my todo list.

these tests as they are written are not exactly elegant… but should work til next changes :D

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
thanks for you contribution @KoffeinKaio :coffee:

**Pull request merged**